frontend/src/app/core/state/in-app-notifications/in-app-notifications.service.ts

```diff
@@ -0,0 +1,144 @@
+  notificationsMarkedRead,
+} from 'core-app/core/state/in-app-notifications/in-app-notifications.actions';
+import {
+  EffectCallback,
+  EffectHandler,
+} from 'core-app/core/state/effects/effect-handler.decorator';
+import { UntilDestroyedMixin } from 'core-app/shared/helpers/angular/until-destroyed.mixin';
+import { ActionsService } from 'core-app/core/state/actions/actions.service';
+
+@EffectHandler
+@Injectable()
+export class InAppNotificationsService extends UntilDestroyedMixin {
+  protected store = new InAppNotificationsStore();
+
+  readonly query = new InAppNotificationsQuery(this.store);
+
+  constructor(
+    readonly injector:Injector,
+    private http:HttpClient,
+    private apiV3Service:APIV3Service,
+    private notifications:NotificationsService,
+    private actions$:ActionsService,
+  ) {
+    super();
+  }
+
+  fetchNotifications(params:Apiv3ListParameters):Observable<IHALCollection<InAppNotification>> {
+    const collectionURL = collectionKey(params);
+
+    return this
+      .http
+      .get<IHALCollection<InAppNotification>>(this.notificationsPath + collectionURL)
+      .pipe(
+        tap((events) => {
+          applyTransaction(() => {
+            this.store.add(events._embedded.elements);
+            this.store.update(({ collections }) => (
+              {
+                collections: {
+                  ...collections,
+                  [collectionURL]: {
+                    ids: events._embedded.elements.map((el) => el.id),
+                  },
+                },
+              }
+            ));
+          });
+        }),
+        catchError((error) => {
+          this.notifications.addError(error);
+          throw error;
+        }),
+      );
+  }
+
+  update(id:ID, inAppNotification:Partial<InAppNotification>):void {
+    this.store.update(id, inAppNotification);
+  }
+
+  modifyCollection(params:Apiv3ListParameters, callback:(collection:ID[]) => ID[]):void {
+    const key = collectionKey(params);
+    this.store.update(({ collections }) => (
+      {
+        collections: {
+          ...collections,
+          [key]: {
+            ...collections[key],
+            ids: [...callback(collections[key]?.ids || [])],
+          },
+        },
+      }
+    ));
+  }
+
+  removeFromCollection(params:Apiv3ListParameters, ids:ID[]):void {
+    const key = collectionKey(params);
+    this.store.update(({ collections }) => (
+      {
+        collections: {
+          ...collections,
+          [key]: {
+            ...collections[key],
+            ids: (collections[key]?.ids || []).filter((id) => !ids.includes(id)),
+          },
+        },
+      }
+    ));
+  }
+
+  markAsRead(notifications:ID[]):Observable<unknown> {
+    return this
+      .apiV3Service
+      .notifications
+      .markRead(notifications)
+      .pipe(
+        tap(() => {
+          this.store.update(notifications, { readIAN: true });
+        }),
+      );
+  }
+
+  private get notificationsPath():string {
```

Comment:
Personal preference, but I like putting getters above the constructor like class properties